crates/matrix-sdk-ui/tests/integration/room_list_service.rs

```diff
@@ -2993,6 +2993,8 @@ async fn test_thread_subscriptions_extension_enabled_only_if_server_advertises_i
                 },
             },
         };
+
+        mock_server.reset().await;
```

Comment:
This plus the change below fix the issue I was seeing locally, consistently making this test fail, and sometimes fail in CI. It seems like there is some weird race condition.